### PR TITLE
TRUNK-6081: Add not null constraint to encounter_diagnosis.dx_rank 

### PR DESCRIPTION
**Pull request by wikumChamith**
## Description of what I changed

Jave field `Diagnosis.rank` is marked as not nullable via the JPA annotations but `encounter_diagnosis.dx_rank` column is missing the not null constraint in the database. So this adds not null constraint to `encounter_diagnosis.dx_rank` column.

## Issue I worked on

https://issues.openmrs.org/browse/TRUNK-6081

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml [hunk 1]
```diff
@@ -116,4 +116,8 @@
 		<addForeignKeyConstraint constraintName="medication_dispense_voided_by_fk" baseTableName="medication_dispense" baseColumnNames="voided_by" referencedTableName="users" referencedColumnNames="user_id"/>
 	</changeSet>
 	
+	<changeSet id="TRUNK-6081-2022-07-10" author="wikumChamith">
+		<addNotNullConstraint tableName="encounter_diagnosis" columnName="dx_rank" columnDataType="int"/>
```
Comment (wluyima) [inline api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:120]: You are missing a comment tag, please include one to describe what the changeset does

**Review by wluyima — COMMENTED** [review]
Added a comment

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml (hunk 1 above)
Comment (wikumChamith) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:121]: I added a comment to describe the changeset in the new commit.

Comment (wikumChamith) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:121]: @wluyima could you review the changes?

Comment (wluyima) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:121]: You are a missing a comment

**Comment by wikumChamith** [comment]
@wluyima I added a comment to describe the changeset.

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml (hunk 1 above)
Comment (wikumChamith) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:121]: > You are a missing a comment

 I added a comment with the new commit.

Comment (wluyima) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:121]: I don't see it

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml [hunk 2]
```diff
@@ -116,4 +116,9 @@
 		<addForeignKeyConstraint constraintName="medication_dispense_voided_by_fk" baseTableName="medication_dispense" baseColumnNames="voided_by" referencedTableName="users" referencedColumnNames="user_id"/>
 	</changeSet>
 	
+	<!-- Adding not null constraint to encounter_diagnosis.dx_rank. TRUNK-6081-->
```
Comment (wikumChamith) [inline api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:119]: @wluyima here is the comment I added.

Comment (wluyima) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:119]: That's not what I mean, I mean a liquibase comment, not an xml comment, sorry if I wasn't clear

Comment (wikumChamith) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:119]: @wluyima I added the necessary liquibase comment.

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml (hunk 1 above)
Comment (wluyima) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.6.x.xml:122]: Thanks! Looks good!

**Comment by wikumChamith** [comment]
@wluyima Are there any additional changes I should make in this PR??

**Comment by wluyima** [comment]
Looks good

**Comment by wikumChamith** [comment]
New PR: https://github.com/openmrs/openmrs-core/pull/4199

**Pull request closed without merging**